great_expectations/datasource/fluent/data_asset/data_connector/data_connector.py

```diff
@@ -118,11 +117,30 @@ def get_data_reference_count(self) -> int:
         """
         pass
 
+    @abstractmethod
+    def get_matched_data_references(self) -> List[Any]:
+        """
+        This interface method returns (e.g., cached) data references that were successfully matched based on "BatchRequest" options.
+
+        Returns:
+            List[Any] -- unmatched data references (type depends on cloud storage environment, SQL DBMS, etc.)
+        """
+        pass
+
+    @abstractmethod
+    def get_matched_data_reference_count(self) -> int:
+        """
+        This interface method returns number of all (e.g., cached) matched data references (useful for diagnostics).
+
+        Returns:
+            int -- number of data references identified
+        """
+        pass
+
```

Comment:
Are there tests for these new methods for each of the data connectors? I don't see them in this PR.

Reply:
@anthonyburdi They do indeed appear in the `FilePathDataConnector` -- that level is sufficient for the subclasses.  Thanks.

Reply:
Do these methods have to implemented?
Should they throw a `NotImplementedError`?